src/btree/bt_handle.c

```diff
@@ -518,11 +518,14 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
 
     btree->modified = false; /* Clean */
 
-    btree->syncing = WT_BTREE_SYNC_OFF; /* Not syncing */
-                                        /* Checkpoint generation */
-    btree->checkpoint_gen = __wt_gen(session, WT_GEN_CHECKPOINT);
-    /* Write generation */
-    btree->write_gen = WT_MAX(ckpt->write_gen, conn->base_write_gen);
+    btree->syncing = WT_BTREE_SYNC_OFF;                           /* Not syncing */
+    btree->checkpoint_gen = __wt_gen(session, WT_GEN_CHECKPOINT); /* Checkpoint generation */
+
+    /*
+     * Initialize to the connection-wide base write generation. Having each session's write
```

Comment:
It's a bit convoluted but here goes:

- During a checkpoint, the metadata file's `write_gen` (which gets restored as the connection `base_write_gen` upon restart) is the maximum of all btree specific `write_gen` in that checkpoint.
- So that means that in the normal case, the `conn->base_write_gen`, is greater than any `write_gen` from the previous run. So the range of write generations that we use doesn't intersect between runs (with the exception of import).

I'll add a bit more detail to this comment.

Reply:
7da9f81